lib/trino-filesystem/src/main/java/io/trino/filesystem/Location.java

```diff
@@ -65,6 +65,10 @@ public static Location parse(String location)
             return new Location(location, Optional.empty(), Optional.empty(), Optional.empty(), OptionalInt.empty(), location.substring(1));
         }
 
+        if (location.startsWith("file:/") && ((location.length() == 6) || (location.charAt(6) != '/'))) {
```

Comment:
Add a comment that this is a special case just for `file:/` and that this follows the expectation of Hadoop